include/dsn/dist/replication/replication_app_base.h

```diff
@@ -106,6 +106,8 @@ class replication_app_base : public replica_base
     // Whether this replica is duplicating.
     bool is_duplicating() const;
 
+    ballot get_ballot() const;
```

Comment:
const ballot& get_ballot() const;

Reply:
In my opinon, that's OK, ballot is just a int64.

Reply:
I don't think so. It's better to use reference to all of the user-defined type. Because it may be changed someday.

Reply:
Or change it to `int64 get_ballot() const;`

Reply:
Done